src/shared/components/oncoprint/DeltaUtils.ts

```diff
@@ -553,10 +563,36 @@ function transitionGenesetHeatmapTrack(
     getTrackSpecKeyToTrackId:()=>{[key:string]:TrackId},
     oncoprint:OncoprintJS<any>,
     nextProps:IOncoprintProps,
-    trackIdForRuleSetSharing:{genesetHeatmap?:TrackId}
+    trackIdForRuleSetSharing:{genesetHeatmap?:TrackId, heatmap?:TrackId}
 ) {
+    function updateExpansionTracks(
+        nextParentSpec: IGenesetHeatmapTrackSpec | undefined,
+        prevParentSpec: IGenesetHeatmapTrackSpec | undefined
+    ) {
+        const expansionTrackList = prevParentSpec ? prevParentSpec.expansionTrackList || [] : [];
```

Comment:
something like how many times transitionHeatmapTrack is called could be tested, not sure if its worth it. i'll leave it to your judgment